### ISSUE TITLE
Communities: Share Selective Account

### ISSUE THREAD

**Issue by FFFra** [status-im/status-react#18010]
# Feature Issue

[Figma Design](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331-288514&mode=design&t=mGXKgDSgRz01Czk4-0)

### User Story

- As a user
- I want to be able to selectively share my account addresses
- So that I can keep some of them private

![Image](https://github.com/status-im/status-mobile/assets/5999878/dacc8490-ec92-444a-bb70-2eb92d93a17f)

### Description

This feature enables users to selectively share their account addresses within the application, enhancing user privacy and control. Users with multiple addresses can choose which ones are visible to others, aligning with various privacy needs and preferences.

### Acceptance Criteria

- Follow Figma
- Users should be able to selectively share individual account addresses, with clear controls for setting each address as public or private.

Depends on: #17977

**Comment by ajayesivan** [comment]
Once [this PR](https://github.com/status-im/status-mobile/pull/18224) is merged, we can integrate the Wallet/Account-Permissions component into this feature. Since this PR requires both code and design review, we can proceed with developing this feature independently without relying on the component. We can later replace the custom implementation with the component, possibly as a follow-up task.
